src/main/java/org/opensearch/opnipreprocessing/plugin/OpniPreProcessor.java

```diff
@@ -91,6 +90,13 @@ public OpniPreProcessor(String tag, String description, String field, String tar
 
     @Override
     public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
+        try {
+            if (isPendingDelete(ingestDocument, nc)) {
+                throw new DeletePendingException(clusterID(ingestDocument));
```

Comment:
Just curious would return `null` works?

Reply:
I don't think it would because it would then ingest the document still.  A runtime exception will avoid ingesting the document but won't crash the process.